cobrand_hackney/static/js.js

```diff
@@ -37,3 +37,64 @@ document.querySelectorAll('.case-detail__map__overlay').forEach(function(b) {
 });
 
 })();
+
+// Creating a new dropdown "Case Locations"
+var caseLocationDiv = document.createElement('div');
+caseLocationDiv.className = 'govuk-form-group lbh-form-group';
+caseLocationDiv.id = 'div_id_case_location';
+caseLocationDiv.innerHTML = '<label for="id_case_location" class="govuk-label lbh-label">Case location</label>' + '<select class="govuk-select lbh-select" id="id_case_location"> <option value="all_areas" selected>All areas</option> <option value="selected_areas">Selected areas</option> <option value="outside_hackney">Outside Hackney</option> </select> ';
+
+// Defining the div that contains all the ward checkboxes
+var area = document.getElementById("div_id_ward");
+
+// Locating "Case Locations" before the area
+parentFormNode = area.parentNode
+parentFormNode.insertBefore(caseLocationDiv, area)
+
+// Defining the rest of the variables
+var inputArea = document.getElementById("id_case_location");
+var outsideHackney = document.getElementById("id_ward_22");
+var myAreasCheckboxChecked = document.querySelectorAll('.govuk-checkboxes__input:checked');
+var myAreasCheckbox = document.querySelectorAll('.govuk-checkboxes__input');
+var inputs = document.querySelectorAll('.govuk-checkboxes__input[name=ward]');
+area.style.display = "none";
+
+// Whenever there is at least one checkbox checked the filter will select by default "selected areas"
+// and will display all the checkboxes.
+if (outsideHackney.checked == true && myAreasCheckboxChecked.length == 1) {
+    inputArea.value = 'outside_hackney';
+} else if (myAreasCheckboxChecked.length == myAreasCheckbox.length) {
+    inputArea.value = 'all_areas';
+} else {
+    inputArea.value = 'selected_areas';
+    area.style.display = "block";
+}
+
+document.querySelector('#id_case_location').addEventListener('change',function(){
+    area.style.display = "none";
+    if (inputArea.value == 'all_areas') {
+        // all the checkboxes for areas will be clicked
+        for(var i = 0; i < inputs.length; i++) {
+            if(inputs[i].type == "checkbox") {
+                inputs[i].checked = true; 
+            }  
+        }
+    } else if (inputArea.value =="outside_hackney") {
+        // Outside hackney
+        for(var i = 0; i < inputs.length; i++) {
+            if(inputs[i].type == "checkbox") {
+                inputs[i].checked = false; 
+            } 
+        }
+        outsideHackney.checked = true
+
+    } else if (inputArea.value == 'selected_areas') { 
+        // selected areas. The checkboxes will become unchecked
```

Comment:
Not sure of best thing to do here - I understand why it would clear everything, but I wonder if it should go back to the 'page load state' (using defaultChecked I guess?) - on page load you might have "your" wards preselected, and if you switch to all/outside and then back to selected, should it blank or go back to "your" wards? Hmm.

Reply:
Thanks for the feedback Matthew. I'll make the changes today =)

Reply:
> 

Yesterday I made the requested changes. However, for that last feedback (defaultChecked), it seems similar to the initial idea of having a fourth option on the dropdown "Case location", called "My areas", that I spoke on the briefing with @zarino. But, we decided not to implement it in this sprint. Let me know if I misunderstood it

Reply:
I didn't know you'd had a discussion :) Fine, we'll see if we get any feedback on it.

Reply:
I completely forgot that, if you have wards associated with your account, they’ll be pre-selected on this page. In that case @lucascumsille, maybe we _can_ actually implement that "My areas" option in the dropdown menu after all. Guess the difficulty will be working out whether, if checkboxes are pre-selected at pageload, that’s because they’re the user’s allocated wards, or they’ve been custom selected by a previous form submission or a ward link from a case detail page. Hmmmm…

Reply:
You could parse the URL and if it doesn't have any `ward=` in it, then whatever is selected at page load must be the user's pre-selected areas (I think).